label-application-rule-config-service-impl/src/main/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleConfigServiceImpl.java

```diff
@@ -47,6 +57,25 @@ public void createLabelApplicationRule(
     try {
       RequestContext requestContext = RequestContext.CURRENT.get();
       this.requestValidator.validateOrThrow(requestContext, request);
+      if (request.getData().getLabelAction().hasDynamicLabelExpression()) {
+        List<LabelApplicationRule> labelApplicationRules =
+            this.labelApplicationRuleStore.getAllObjects(requestContext).stream()
+                .map(ConfigObject::getData)
+                .collect(Collectors.toUnmodifiableList());
+        int dynamicLabelApplicationRules = 0;
+        for (LabelApplicationRule rule : labelApplicationRules) {
+          if (rule.getData().getLabelAction().hasDynamicLabelExpression()) {
+            dynamicLabelApplicationRules = dynamicLabelApplicationRules + 1;
+          }
+        }
+        if (dynamicLabelApplicationRules >= maxLabelApplicationRuleAllowed) {
```

Comment:
Should be `>` check only

Reply:
If the create request wants to create a rule with dynamic label then the count of already existing dynamic label application rules must be strictly less than the max_allowed limit. right?... hence the `>=` condition.

Reply:
Makes sense. Missed the current rule part.